examples/preprocessing/plot_target_encoder_cross_val.py

```diff
@@ -0,0 +1,159 @@
+"""
+==========================================
+Target Encoder's Internal Cross Validation
+==========================================
+
+.. currentmodule:: sklearn.preprocessing
+
+The :class:`TargetEnocoder` replaces each category of a categorical feature with
+the mean of the target variable for that category. This method is useful
+in cases where there is a strong relationship between the categorical feature
+and the target. To prevent overfitting, :meth:`TargetEncoder.fit_transform` uses
+interval cross validation to encode the training data to be used by a downstream
+model. In this example, we demonstrate the importance of the cross validation
+procedure to prevent overfitting.
+"""
+
+# %%
+# Create Synthetic Dataset
+# ========================
+# For this example, we build a dataset with three categorical features: an informative
+# and feature with medium cardinality, an uninformative feature with medium cardinality,
+# and an uninformative feature with high cardinality. First, we generate the informative
+# feature:
+from sklearn.preprocessing import KBinsDiscretizer
+import numpy as np
+
+n_samples = 50_000
+
+rng = np.random.RandomState(42)
+y = rng.randn(n_samples)
+noise = 0.5 * rng.randn(n_samples)
+n_categories = 100
+
+kbins = KBinsDiscretizer(
+    n_bins=n_categories, encode="ordinal", strategy="uniform", random_state=rng
+)
+X_informative = kbins.fit_transform((y + noise).reshape(-1, 1))
+
+# Permute the feature to remove the information from the ordering
```

Comment:
I took me a while to figure out what this meant. How about the following: "Remove the linear relationship between y and the bin index by permuting the values of X_informative" (or some such)?